js-webapp-async/README.md

```diff
@@ -1,22 +1,26 @@
 
-Notice that we also added a slider input to the page. This slider isn't connected to anything, and is there to illustrate UI blocking. Notice that while the webassembly code is still running, the slider is completely unresponsive. If this were an actual webapp and not just a demo, the UI blocking would surely annoy any users of the system and possibly make working with the app cumbersome and impracticle. (TODO: animated gif of slider not responding) We can easily solve this, and keep the UI responsive at all times, using web workers.
+Notice that we also added a slider input to the page. This slider isn't connected to anything, and is there just to illustrate UI blocking. Notice that while the webassembly code is still running, the slider is completely unresponsive. If this were an actual webapp and not just a demo, the UI blocking would surely annoy any users of the system and possibly make working with the app cumbersome and impracticle. We can easily solve this, and keep the UI responsive at all times, using web workers.
+
+![blocking ui](blocking.gif)
+
+Blocked ui while code is running
 
 ### Web workers
 
-A [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) is an object that handles execution of a piece of code aynchronously. On completion the web worker will notify so the result can be processed and, in this case, rendered on the page. 
+A [web worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) is an object that handles execution of a piece of code in another thread. On completion the web worker will notify so the result can be processed and, in this case, rendered on the page. 
 
 
-The way the page communicates with the worker object is through sending messages. The page will send a message to the worker to start doing work, and the message will include all data that the worder needs. The worker will perform its execution, using only the data that was in the message. The worker will not be able to access any data in the webapp directly. The worker can only send back results by sending a message back to the webapp. 
+The way the page communicates with the worker object is through sending messages. The page will send a message to the worker to start doing work, and the message will include all data that the worder needs. The worker will perform its execution, using only the data that was in the message, as the worker will not be able to access any data in the webapp directly. When finished the worker needs to communicate the results back to the webapp. It can do this by sending a message. 

```

Comment:
```suggestion
The way the page communicates with the worker object is through sending messages. The page will send a message to the worker to start doing work, and the message will include all data that the worker needs. The worker will perform its execution, using only the data that was in the message, as the worker will not be able to access any data in the web app directly. When finished the worker needs to communicate the results back to the web app. It will do this by sending a message. 

```